js/iframe-coordinator/elements/frame-router.ts

```diff
@@ -21,9 +33,18 @@ class FrameRouterElement extends HTMLElement {
 
   registerClients(clients) {
     this.router = Host.embed(this, clients);
+    this.router.ports.toHost.subscribe(message => {
+      if (message.msgType == "publish") {
+        let event = new CustomEvent("publish", { detail: message.msg });
+        this.dispatchEvent(event);
+      }
+    });
```

Comment:
When the host library receives a publication from the client, it checks within Elm to see if the host is subscribed, if so, the event is emitted on the `toHost` port and dispatched as a custom DOM event here.